tooling/minimize/src/rvalue.rs

```diff
@@ -170,12 +171,18 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
             smir::Rvalue::CopyForDeref(place) =>
                 ValueExpr::Load { source: GcCow::new(self.translate_place_smir(place, span)) },
             smir::Rvalue::Len(place) => {
-                // as slices are unsupported as of now, we only need to care for arrays.
                 let ty = place.ty(&self.locals_smir).unwrap();
-                let Type::Array { elem: _, count } = self.translate_ty_smir(ty, span) else {
-                    panic!()
-                };
-                ValueExpr::Constant(Constant::Int(count), <usize>::get_type())
+                match self.translate_ty_smir(ty, span) {
+                    Type::Array { elem: _, count } =>
+                        ValueExpr::Constant(Constant::Int(count), <usize>::get_type()),
```

Comment:
```suggestion
                        // FIXME: still evaluate the place -- it might have UB after all.
                        ValueExpr::Constant(Constant::Int(count), <usize>::get_type()),
```